### PR TITLE
Upgrade DPS Sdk to V20180901

### PR DESCRIPTION
**Pull request by anusapan**
Add reprovision-policy to the enrollment and enrollment-group.
Add symmetric key attestation to the enrollment.
Upgrade dps sdk to v20180901.
Add new dps unit test and integration test
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [x] Have all unit and integration tests passed locally? i.e. `pytest -s <project root>/azext_iot/tests`
- [ ] Have static checks passed using the .pylintrc and .flake8 rules?

### CODE REVIEW

**Comment by anusapan** [comment]
Closed the previous PR and created this with cleaned commit history

File: azext_iot/dps_sdk/device_provisioning_service_service_runtime_client.py [hunk 1]
```diff
@@ -33,12 +36,12 @@ def __init__(
         if credentials is None:
             raise ValueError("Parameter 'credentials' must not be None.")
         if not base_url:
-            base_url = 'https://<fully-qualified IoT hub domain name>'
+            base_url = 'https://localhost'
 
         super(DeviceProvisioningServiceServiceRuntimeClientConfiguration, self).__init__(base_url)
 
         self.add_user_agent('deviceprovisioningserviceserviceruntimeclient/{}'.format(VERSION))
-        self.add_user_agent('MicrosoftAzure/IoTPlatformCliExtension/{}'.format(extver))
```
Comment (digimaun) [inline azext_iot/dps_sdk/device_provisioning_service_service_runtime_client.py]: Please keep the custom user agent for downstream usage metrics

File: azext_iot/dps_sdk/device_provisioning_service_service_runtime_client.py [hunk 2]
```diff
@@ -11,9 +15,8 @@
 from .version import VERSION
 from .operations.device_enrollment_operations import DeviceEnrollmentOperations
 from .operations.device_enrollment_group_operations import DeviceEnrollmentGroupOperations
-from .operations.registration_status_operations import RegistrationStatusOperations
+from .operations.registration_state_operations import RegistrationStateOperations
 from . import models
-from azext_iot._constants import VERSION as extver
```
Comment (digimaun) [inline azext_iot/dps_sdk/device_provisioning_service_service_runtime_client.py]: import for combining version for user agent

Comment (anusapan) [reply azext_iot/dps_sdk/device_provisioning_service_service_runtime_client.py]: change reverted

File: azext_iot/dps_sdk/device_provisioning_service_service_runtime_client.py (hunk 1 above)
Comment (anusapan) [reply azext_iot/dps_sdk/device_provisioning_service_service_runtime_client.py]: change reverted

**Pull request merged**